src/dcegm/fast_upper_envelope.py

```diff
@@ -825,57 +668,555 @@ def _backward_scan(
     )
 
 
-def _evaluate_point_on_line(
-    x1: float, y1: float, x2: float, y2: float, point_to_evaluate: float
-) -> float:
-    """Evaluate a point on a line.
+def update_bools_and_idx_to_inspect(idx_to_inspect, update_idx, case_2, case_5):
```

Comment:
Can you check whether the docstring and arg definitions here are correct?

Reply:
Yes, why not?